core/src/main/java/org/apache/calcite/util/NlsString.java

```diff
@@ -227,6 +227,27 @@ public NlsString rtrim() {
     return this;
   }
 
+  /**
+   * Returns a string the same as this but with spaces trimmed from the
+   * right such that the result has the specified size.
+   *
+   * @param resultSize Expected size for result string.  If negative, it indicates
+   *                   that no trimming should be done.
+   */
+  public NlsString rtrim(int resultSize) {
```

Comment:
@mihaibudiu The implementation is a bit confusing: For example:
```
NlsString string = new NlsString("hello world ", null, null);
string.rtrim(5);
```
produces "helloworld". It removes spaces from the middle, even though the documentation claims it should remove them only from the right.

Reply:
Good catch, should call rtrim on the fragment, not trim.

Reply:
Sorry, pushed one more correction, hopefully now it makes more sense.